bench/BenchAC.hs

```diff
@@ -32,115 +40,171 @@ evalBS (ByteString xs) = eval xs M.empty
+        , benchCompilation @F "Hash circuit" "compiled_scripts/hash_circuit_128.json" hc128
+        , benchCompilation @F "Hash circuit" "compiled_scripts/hash_circuit_256.json" hc256
+        , benchCompilation @F "Hash circuit" "compiled_scripts/hash_circuit_512.json" hc512
+        ]
+
+-- mainHash :: IO ()
+-- mainHash = do
+--   getCurrentTime >>= print
+--   ByteString ac32  <- hashCircuit @32 @BLS12_381_Scalar
+--   getCurrentTime >>= print
+--   ByteString ac64  <- hashCircuit @64 @BLS12_381_Scalar
+--   getCurrentTime >>= print
+--   ByteString ac128 <- hashCircuit @128 @BLS12_381_Scalar
+--   getCurrentTime >>= print
+--   ByteString ac256 <- hashCircuit @256 @BLS12_381_Scalar
+--   getCurrentTime >>= print
+--   ByteString ac512 <- hashCircuit @512 @BLS12_381_Scalar
+--   getCurrentTime >>= print
+
+--   putStrLn "Sizes"
+
+--   print $ acSizeM ac32
+--   getCurrentTime >>= print
+--   print $ acSizeM ac64
+--   getCurrentTime >>= print
+--   print $ acSizeM ac128
+--   getCurrentTime >>= print
+--   print $ acSizeM ac256
+--   getCurrentTime >>= print
+--   print $ acSizeM ac512
+--   getCurrentTime >>= print
+
+--   putStrLn "Evaluation"
+
+--   print $ exec ac32
+--   getCurrentTime >>= print
+--   print $ exec ac64
+--   getCurrentTime >>= print
+--   print $ exec ac128
+--   getCurrentTime >>= print
+--   print $ exec ac256
+--   getCurrentTime >>= print
+--   print $ exec ac512
+--   getCurrentTime >>= print
+
+--   defaultMain
+--       [ benchCircuit "SHA2 512/364" $ hashCircuit @32 @BLS12_381_Scalar
+--       , benchCircuit "SHA2 512/364" $ hashCircuit @64 @BLS12_381_Scalar
+--       , benchCircuit "SHA2 512/364" $ hashCircuit @128 @BLS12_381_Scalar
+--       , benchCircuit "SHA2 512/364" $ hashCircuit @256 @BLS12_381_Scalar
+--       , benchCircuit "SHA2 512/364" $ hashCircuit @512 @BLS12_381_Scalar
+--       ]
+
+-- mainSumBS :: IO ()
+-- mainSumBS = do
+
+--   printCircuitSize @BLS12_381_Scalar @32 additionCircuit
+--   printCircuitSize @BLS12_381_Scalar @64 additionCircuit
+--   printCircuitSize @BLS12_381_Scalar @128 additionCircuit
+--   printCircuitSize @BLS12_381_Scalar @256 additionCircuit
+--   printCircuitSize @BLS12_381_Scalar @512 additionCircuit
+
+--   defaultMain
+--       [ benchCircuit "Adding ByteStrings" $ additionCircuit @32 @BLS12_381_Scalar
+--       , benchCircuit "Adding ByteStrings" $ additionCircuit @64 @BLS12_381_Scalar
+--       , benchCircuit "Adding ByteStrings" $ additionCircuit @128 @BLS12_381_Scalar
+--       , benchCircuit "Adding ByteStrings" $ additionCircuit @256 @BLS12_381_Scalar
+--       , benchCircuit "Adding ByteStrings" $ additionCircuit @512 @BLS12_381_Scalar
+--       ]
+
+-- mainSumWitnnessGeneration :: IO ()
+-- mainSumWitnnessGeneration = do
+
+--   defaultMain
+--       [ benchWitnessGeneration "Adding ByteStrings" $ toFieldElements @(ArithmeticCircuit F) <$> additionCircuit @32 @BLS12_381_Scalar
```

Comment:
Maybe we can include witness generation into the benchmark as well, just to make it be seen in case something goes wrong?

Reply:
The general idea is to have separate benchmarks for compilation and evaluation that would share a collection of functions on which we run them.

Reply:
So I guess the benchmarks have to be split in different modules? How should this work with the cabal's benchmarking framework